onnxruntime/core/framework/orttraining_partial_executor.h

```diff
@@ -19,8 +19,15 @@
 namespace onnxruntime {
 class PartialExecutor : public IExecutor {
  public:
-  PartialExecutor(PartialGraphExecutionState& state)
-      : state_{state} {}
+  PartialExecutor(PartialGraphExecutionState& state,
+                  const OrtValueCache& cache)
+      : state_{state},
+        cache_{cache} {
+    if (cache.empty()) {
+      // To avoid checking the cache for each node if it is empty
+      check_cache_ = false;
+    }
```

Comment:
No need to create an additional variable, please directly using cache.empty() becasue its complexity is constant, ref: [https://www.cplusplus.com/reference/unordered_map/unordered_map/empty/](https://www.cplusplus.com/reference/unordered_map/unordered_map/empty/)

Reply:
see https://github.com/microsoft/onnxruntime/pull/8273#issuecomment-900515827